integreat_cms/cms/views/pages/partial_page_tree_view.py

```diff
@@ -1,61 +1,74 @@
-from django.shortcuts import render
-from django.utils.decorators import method_decorator
-from django.views.generic import TemplateView
+from collections import defaultdict
+import json
 
-from integreat_cms.cms.views.pages.page_context_mixin import PageContextMixin
+from django.http import JsonResponse
+from django.template.loader import render_to_string
+from django.utils.translation import get_language
+from django.views.decorators.http import require_POST
 
+from ..pages.page_context_mixin import PageContextMixin
+from ...models.languages.language import Language
 from ...decorators import permission_required
 
 
-@method_decorator(permission_required("cms.view_page"), name="dispatch")
-class PartialPageTreeView(TemplateView, PageContextMixin):
-    """
-    View for rendering a partial page tree
-    """
+@permission_required("cms.view_page")
+@require_POST
+# pylint: disable=unused-argument
+def render_partial_page_tree_views(request, region_slug, language_slug):
+    r"""
+    Retrieve the rendered subtree of a given root page
 
-    #: Template for a partial page tree
-    template = "pages/_page_tree_children.html"
+    :param request: The current request
+    :type request: ~django.http.HttpRequest
 
-    # pylint: disable=unused-argument
-    def get(self, request, *args, **kwargs):
-        r"""
-        Retrieve the rendered subtree of a given root page
+    :param region_slug: The slug of the current region
+    :type region_slug: str
 
-        :param request: The current request
-        :type request: ~django.http.HttpRequest
+    :param language_slug: The slug of the current language
+    :type language_slug: str
 
-        :param \*args: The supplied arguments
-        :type \*args: list
+    :return: The rendered template responses
+    :rtype: ~django.http.JsonResponse
+    """
+    requested_tree_ids = [int(i) for i in json.loads(request.body.decode("utf-8"))]
 
-        :param \**kwargs: The supplied keyword arguments
-        :type \**kwargs: dict
+    region = request.region
+    language = region.get_language_or_404(language_slug, only_active=True)
 
-        :return: The rendered template response
-        :rtype: ~django.template.response.TemplateResponse
-        """
-        tree_id = int(kwargs.get("tree_id"))
-        region = request.region
-        language = region.get_language_or_404(
-            kwargs.get("language_slug"), only_active=True
-        )
+    backend_language = Language.objects.filter(slug=get_language()).first()
+
+    all_pages = (
+        region.pages.filter(tree_id__in=requested_tree_ids)
+        .prefetch_major_public_translations()
+        .prefetch_related("mirroring_pages")
+        .cache_tree(False)
```

Comment:
```suggestion
        .cache_tree(archived=False)
```
I'd write the kwarg explicitly here to make it more obvious what is happening.